sawmill-core/src/main/java/io/logz/sawmill/processors/ListIntersectProcessor.java

```diff
@@ -0,0 +1,85 @@
+package io.logz.sawmill.processors;
+
+import io.logz.sawmill.Doc;
+import io.logz.sawmill.ProcessResult;
+import io.logz.sawmill.Processor;
+import io.logz.sawmill.annotations.ProcessorProvider;
+import io.logz.sawmill.utilities.JsonUtils;
+
+import javax.inject.Inject;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+import static org.apache.commons.collections4.CollectionUtils.intersection;
+import static org.apache.commons.collections4.CollectionUtils.isNotEmpty;
+
+@ProcessorProvider(type = "listIntersect", factory = ListIntersectProcessor.Factory.class)
+public class ListIntersectProcessor implements Processor {
```

Comment:
I think it's better to call it `arraysIntersect` and `ArraysIntersectProcessor`, this way it's aligned with ElasticSearch type names. (there is no list in ES / OS). 

plural form is probably better because we need 2 arrays to do the intersect

Reply:
Shit. I missed this comment @matvey-mtn.